lib/luanetfilter.c

```diff
@@ -99,18 +190,33 @@ static int luanetfilter_xtable(lua_State *L)
 
 	luaxtable->runtime = lunatik_toruntime(L);
 	lunatik_getobject(luaxtable->runtime);
+
+	luaL_checktype(L, 1, LUA_TTABLE);
+
+	/* handle match and target */
+	lua_getfield(L, 1, "match");
+	if(handle_xtable_reg(L, luaxtable, LUANETFILTER_MATCH) != 0 )
+		luaL_error(L, "[luanetfilter] handle match error - %s\n", lua_tostring(L, -1));
+
+	lua_getfield(L, 1, "target");
+	if(handle_xtable_reg(L, luaxtable, LUANETFILTER_TARGET) != 0 )
```

Comment:
it's not clear to me why you are using this enum instead of just passing the string parameter around.. e.g.,
```
luanetfilter_xtable_register(L, luaxtable, "target");
```

Reply:
In the new commits the if..else block need the enums, otherwise I can use bool too, if that is better ?